### DISCUSSION TITLE
How to set up a public server for windows? (easy on Mac!)

### DISCUSSION THREAD

**Discussion by phaves** [jamulussoftware/jamulus#1903] [Technical Support & Help] [answered]
When I install Jamulus on a Mac I get two executables: Jamulus and JamulusServer. JamulusServer makes it very easy to set up a public server but I can't see how to do this under Windows - there appears to be no JamulusServer.exe or equivalent. What should I do? How can Jamulus documentation be improved in this regard, or am I missing something (after many hours searching the documentation)?

Thanks, Philip

**Comment by dcorson-ticino-com** [comment]
I don't know how this is packaged on a Mac, but there is only one Jamulus program.  That is called with different command line parameters  to run as client or server.
When I install on Win10 I get two shortcuts in the program list, one for Server and one for Client. The server shortcut includes the -s command line option to cause Jamulus to run as a server.  There is, however, only one .exe.

**Reply by gene96817** [reply]
Most Mac users see the install package as delivering two apps. The underlining command lines are not in their worldview.

**Comment by henkdegroot** [comment, marked as answer]
> or am I missing something

I am sue you have looked at this page: [https://jamulus.io/wiki/Server-Win-Mac](https://jamulus.io/wiki/Server-Win-Mac)

In which the first line says: **Windows users** - Start the server using the entry for “Jamulus server” in the Windows start menu.

Once you have started the Jamulus as server, you can even get to the "Options" tab and enable the "Start Minimized on Windows Start" option. This will then automatically start the server after booting windows and login.

**Comment by phaves** [comment]
Dear Henk,

Thanks for you (and others) for your help. I am now up and running.

I've been trying to understand the origin of my difficulties in setting 
up a Public server under Windows, and whether the documentation could be 
easier to navigate.

First, the Mac installation produces two executables, each with an icon 
on the task bar - very clear and easy. Regarding Windows, I had seen 
statements from users indicating that the installation produces two 
shortcuts, one each for Jamulus and Jamulus Server; this is not the case 
in the current version. I had also seen statements that Jamulus server 
could be invoked from the command line; unfortunately 
https://jamulus.io/wiki/Command-Line-Options 
<https://jamulus.io/wiki/Command-Line-Options> does not include -s  - it 
would be helpful to add this, near the top of the page.

The initial documentation I found: Setup - Jamulus Wiki and Software 
Manual - Jamulus Wiki (understandably) did not address my question. My 
major mistake was in thinking that Running a Server | Installation for 
Windows and macOS was for setting up a private server and so not 
visiting that page because I thought (erroneously) that it didn't apply 
to me.

I appreciate that writing documentation involves a fine balance between 
conciseness and redundancy and that anticipating all the different paths 
that users might take is very difficult, so I am very appreciative of 
all the effort that has gone into the production of the documentation.

Best wishes,

Philip Haves


On 7/8/2021 12:07 PM, Henk De Groot wrote:
>
>     or am I missing something
>
> I am sue you have looked at this page: 
> https://jamulus.io/wiki/Server-Win-Mac 
> <https://jamulus.io/wiki/Server-Win-Mac>
>
> In which the first line says: *Windows users* - Start the server using 
> the entry for “Jamulus server” in the Windows start menu.
>
> Once you have started the Jamulus as server, you can even get to the 
> "Options" tab and enable the "Start Minimized on Windows Start" 
> option. This will then automatically start the server after booting 
> windows and login.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/jamulussoftware/jamulus/discussions/1903#discussioncomment-981031>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ACJV7I7MKEFKKVG3G6WIX6TTWXZOFANCNFSM477UHSDA>.
>
--

**Reply by dcorson-ticino-com** [reply]
You have a very valid point, it looks like the -s option somehow slipped through the cracks of the last documentation update.  I don't see it either.

On the other hand I just uninstalled Jamulus on my Win10 compi, downloaded 3.8.0 and installed it and now have Jamulus and Jamulus Server as shortcuts in the start-up menu.  I don't know what is happening on your system.

**Reply by ann0see** [reply]
Hi @phaves giving feedback like that is very important! Thank you very much. 

@gilgongo already worked on a new server manual ;-). You‘re more than welcome to have a look at the new (unpublished) files: see e.g. https://github.com/jamulussoftware/jamuluswebsite/blob/next-release/wiki/en/Running-a-Server.md 

…and the server related files here: https://github.com/jamulussoftware/jamuluswebsite/tree/next-release/wiki/en
